src/components/calendar/WeekCalendar/WeekCalendarEvent.tsx

```diff
@@ -0,0 +1,117 @@
+
+    const endOfDay = new Date(new Date(startOfDay)
+        .setDate(startOfDay.getDate() + 1));
+
+    const naiveStartDate = new Date(event.start_time);
+    const naiveEndDate = new Date(event.end_time);
+
+    const startTime = new Date(naiveStartDate.getUTCFullYear(), naiveStartDate.getUTCMonth(), naiveStartDate.getUTCDate(), naiveStartDate.getUTCHours(), naiveStartDate.getUTCMinutes());
+
+    const endTime = new Date(naiveEndDate.getUTCFullYear(), naiveEndDate.getUTCMonth(), naiveEndDate.getUTCDate(), naiveEndDate.getUTCHours(), naiveEndDate.getUTCMinutes());
+
+    const startsBeforeToday = startTime <= startOfDay;
+    const endsAfterToday = endTime >= endOfDay;
+
+    const getEventPos = () => {
+        const oneMinute = 100 / 1440;
+
+        const startFromMidnight = startsBeforeToday ? 0 : (startTime.getTime() - new Date(startTime).setHours(0, 0, 0, 0)) / 60000;
+
+        const endFromMidnight = endsAfterToday ? 24 * 60 : (endTime.getTime() - new Date(endTime).setHours(0, 0, 0, 0)) / 60000;
+        const diff = endFromMidnight - startFromMidnight;
+
+        return {
+            height: `calc(${diff * oneMinute}% - 2px)`,
+            top: `${startFromMidnight * oneMinute}%`,
+        };
+    };
+
+    const getEventTimeLabel = () => {
+        return (
+            <>
+                <FormattedTime value={ new Date(startsBeforeToday? startOfDay: startTime) }/>{ ` - ` }
+            </>
+        );
+    };
+
+    return (
+        <li>
+            <NextLink href={  baseHref + `/calendar/events/${event.id}` } passHref>
+                <Link>
+                    <Box
+                        { ...( focussed && { zIndex: 10 } ) }
+                        display="flex"
+                        height={ getEventPos().height }
+                        justifyContent="flex-end"
+                        position="absolute"
+                        top={ getEventPos().top }
+                        width={ 1 }>
+                        <Tooltip arrow placement="left" title={ event.title || event.activity.title }>
+                            <Box
+                                { ...{ ref: eventDiv } }
+                                bgcolor={ (campaign?.color
+                                    || DEFAULT_COLOR) + `${focussed? '': '55'}` }
+                                borderLeft={ `5px solid ${campaign?.color || DEFAULT_COLOR}` }
+                                color={ getContrastColor(campaign?.color || DEFAULT_COLOR) }
+                                data-testid={ `event-${event.id}` }
+                                height={ 1 }
+                                onMouseEnter={ () => setFocussed(true) }
+                                onMouseLeave={ () => setFocussed(false) }
+                                padding={ 1 }
+                                width={ `${100 - 5 * (shiftValue || 0)}%` }>
```

Comment:
I think maybe this width should be on the top-level container instead? Currently, I'm experiencing a bug with these two events:

![image](https://user-images.githubusercontent.com/550212/125341822-5ce1ec00-e354-11eb-8d51-33903a6adc6b.png)

I can't seem to hover the rearmost event, probably because the `li`, `NextLink`, `Link`, outer `Box` or `Tooltip` components steal the `mouseover` event before it reaches the innermost `Box`.